### PR TITLE
refac: Use offset from SDK for executable loaders

### LINKED ISSUE #6683: Use offset from SDK for executable loaders instead of re-calculating it
## Description

Loader offset is exposed by sdk as of https://github.com/FuelLabs/fuels-rs/pull/1522 currently we are recalculating this value in forc-deploy because at the time of the implementation this was not exposed by sdk. This is bad because if anything changes around that area, we need to update it manually. We should use the exposed value from sdk instead of recalculating.

https://github.com/FuelLabs/fuels-rs/pull/1522

### Context
-------
https://github.com/FuelLabs/sway/blob/b6bbbf8bf5939eaf2c965b4b94732d43ef656500/forc-plugins/forc-client/src/op/deploy.rs#L357-L360

## Solution

Basically remove the function showed in context, and get the exposed value [from SDK](https://github.com/FuelLabs/fuels-rs/pull/1522) directly.

### PR DESCRIPTION
**Pull request by Abeeujah**
## Description
Uses the offset exposed by the SDK instead of recalculating it and having to manually update the code each time the implementation changes.

closes #6683 

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Comment by Abeeujah** [comment]
Hi @kayagokalp @Br1ght0ne @Salka1988 @hal3e please can you review?

File: forc-plugins/forc-client/src/op/deploy.rs [hunk 1]
```diff
@@ -482,28 +382,25 @@ pub async fn deploy_executables(
...
+                    .into_iter()
+                    .map(|config| Configurable {
+                        offset: config.offset - offset_shift as u64,
+                        ..config.clone()
+                    })
+                    .collect()
+            });
+            fuel_abi.configurables = configurables;
+            let json_string = serde_json::to_string_pretty(&fuel_abi)?;
+            std::fs::write(json_abi_path, json_string)?;
```
Comment (kayagokalp) [inline forc-plugins/forc-client/src/op/deploy.rs:403]: Looks good to me, but I believe we need a test for this. Especially because we are not sure what is going to happen if the binary does not have a `data` section. In earlier implementation it was obvious that the function to calculate offset returns `None` in that case but now I am not sure what is going to happen. 

So if you are up for it, let's add a test which get's its oracle values (like the calculated shifted values with the previous implementation) and in a next commit let's apply this change and see if the calculated values are same.

Comment (kayagokalp) [inline forc-plugins/forc-client/src/op/deploy.rs:403]: Looks good to me, but I believe we need a test for this. Especially because we are not sure what is going to happen if the binary does not have a data section. In earlier implementation it was obvious that the function to calculate offset returns None in that case but now I am not sure what is going to happen.

So if you are up for it, let's add a test which get's its oracle values (like the calculated shifted values with the previous implementation) and in a next commit let's apply this change and see if the calculated values are same.

**Review by kayagokalp — CHANGES_REQUESTED** [review]
Looks great! Thanks for the contribution. Can we add a test maybe? I left the details in the comment 🚀

File: forc-plugins/forc-client/src/op/deploy.rs (hunk 1 above)
Comment (Abeeujah) [reply forc-plugins/forc-client/src/op/deploy.rs:403]: Hi @kayagokalp I did look at the SDK before I implemented this change, the function implementation is guaranteed to always return an offset, and cases where it wouldn't the program crashes here's a snippet from `Executable<Loader>` impl block
```rust
pub fn data_offset_in_code(&self) -> usize {
        self.code_with_offset().1
    }

    fn code_with_offset(&self) -> (Vec<u8>, usize) {
        let mut code = self.state.code.clone();

        self.state.configurables.update_constants_in(&mut code);

        let blob_id = self.blob().id();

        transform_into_configurable_loader(code, &blob_id)
            .expect("checked before turning into a Executable<Loader>")
    }
```

But if you'd want a test after looking at this, what should I be testing for? that the value isn't zero? within a specific range? I'd like to hear your POV

Comment (kayagokalp) [reply forc-plugins/forc-client/src/op/deploy.rs:403]: Hey again, it is possible that a binary does not contain any data section, that happens if the sway code does not have any constants and no configurable section. So if SDK is expecting this to be checked before calling the function, we might need to check this explicitly. 

For testing we can use a example test script or predicate's binary from `/test/data`. And (possibly) extract the shift calculation to a separate function so that we can write tests against it.

Basically call this new calculate shift function and check if the calculated shift is being calculated correctly for two types of sway scripts:

1. A script that has a single main function and nothing else.
2. The script like https://github.com/FuelLabs/sway/blob/master/forc-plugins/forc-client/test/data/deployed_script/src/main.sw which needs this shift.

I can help with testing if you need it feel free to ping me

Comment (Abeeujah) [reply forc-plugins/forc-client/src/op/deploy.rs:403]: Yeah, I'd need some help with testing it, How do I ping you? I tried on twitter, 2 days ago and you haven't responded

Comment (kayagokalp) [reply forc-plugins/forc-client/src/op/deploy.rs:403]: I will be adding the tests to this PR directly to better streamline the process

**Comment by sdankel** [comment]
@Abeeujah  Thank you for your contribution, but this PR is superseded by https://github.com/FuelLabs/sway/pull/6991 since in future versions we will no longer use the data section offset.
